src/transformState.js

```diff
@@ -1,11 +1,34 @@
+/* eslint-disable spaced-comment */
 'use strict';
 
 /**
  * @param {Object} state
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  const reducer = (reducerState, action) => {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(reducerState, action.extraData);
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete reducerState[key];
+        }
+        break;
+      case 'clear':
+        for (const param in reducerState) {
+          delete reducerState[param];
+        }
+        break;
+      default:
+        break;
+    }
+  };
```

Comment:
This helper function also can be moved outside `transformState` function in order to make it reusable